### DISCUSSION TITLE
[Suggestion] Spotify-like Heart button

### DISCUSSION THREAD

**Discussion by azagthoth1** [TT-ReBORN/Georgia-ReBORN#169] [Ideas]
I wrongly posted this as an Issue, since I didn't know there was a specific place for this. I apologize. I'll just copy and paste what I wrote before. 

This theme is the BEST there ever was, there ever is and there will ever be, however, I feel like implementing some sort of feature to "like" or favorite songs playing and directly send them/save them to an automatically created playlist (let's just call it "Favorites"), would be the cherry on top.

Sorry if what I'm asking is too much, I know that the code is open source and I could try to do it myself, If I was a programmer, of course, but that is not the case. Could you guys consider giving it a little thought, so maybe it could be added in a future release? Would that be possible to implement?

I can't create a mockup, but it could be added to the bottom bar, between the playback buttons, perhaps even on the left side of the stop button. It would all be symmetrical, with the play button in the middle. I'd lose my shit. Thank you.

**Comment by TT-ReBORN** [comment]
Hi @azagthoth1,

glad you like the theme! 
I like your idea and the best solution would be to implement an optional lower bar button ( like the reload button ) 
that has this functionality. Furthermore the user could configure and set the target playlist in which those songs will be added.

I am currently refactoring the code and structure of Georgia-ReBORN which will be much better but this master task 
needs to be finished first. I will let you know when I have implemented this button here.

-TT

**Reply by azagthoth1** [reply]
Wow, I'm happy that you liked it! Now I just had another idea to add to this. What if we could also have the option to add a "Playlist" button to the top bar [1], where we also have Details, Library, Biography, Lyrics and Rating? That way, I guess, once we start liking the songs, we could just click on the tab and have a view of them, not that much different from when we have a view "inside" [2] (for the lack of better term, since English is not my native language) the album we're playing when we click on Library again from the "Album-view mode".

[1] ![image](https://github.com/TT-ReBORN/Georgia-ReBORN/assets/35978048/602e0832-bdcc-4791-9a4a-686583b028ca)

[2] 
![image](https://github.com/TT-ReBORN/Georgia-ReBORN/assets/35978048/c07e84ca-ed0f-4d28-b932-18880b2eb9e4)

**Reply by TT-ReBORN** [reply]
I do not understand your second request clearly, but we already have two buttons for listing all playlists as a drop down menu:
- The first is the hidden mouse hover playlist manager text button which you can find if you mouse hover at the top of the playlist.
- The second is in the top menu > Playlists.

There is also the Library/Playlist split mode which you can drag and drop albums/tracks directly to your playlists.
You can find it if you right click for context menu in the Library > Change layout to split.

-TT

**Comment by TT-ReBORN** [comment]
@azagthoth1,

I have implemented a new "AddTracks" button in the latest commit: https://github.com/TT-ReBORN/Georgia-ReBORN/commit/3c60e2e2d5aa953856a267f8ae350d9599d77572

Here is the excerpt from the changelog:

> - Added a new "Add tracks" button (+) that adds selected albums/tracks from Library or Playlist to user defined playlist.
>   - Can be displayed via top menu Options > Player controls > Lower bar > Show add tracks button
>     or right click on lower bar for context menu > Display > Show add tracks button.
>   - The default playlist for using the add tracks button is "Favorites", the add tracks button playlist can
>     be changed via top menu Options > Player controls > Lower bar > Add tracks button > Add tracks playlist
>     or right click on lower bar for context menu > Controls > Add tracks playlist.
>   - When adding albums/tracks from the Library or Playlist (to another playlist), you can activate additionally
>     top menu Options > Player controls > Panel > Switch to playlist when adding songs
>     or right click on lower bar for context menu > Controls > Switch to playlist when adding songs.

-TT

**Reply by azagthoth1** [reply]
Hello, I hope you are doing good. I only created this account to make this suggestion, because I'm not a programmer or anything. I was thankful that you liked my idea and implemented it, but I thought it would eventually come out in an update, but I just realized that it never did. So, am I supposed to figure out how to build and compile this specific "snapshot" (for lack of better term), if I want to use it before a new version comes out?

**Reply by TT-ReBORN** [reply]
Hi @azagthoth1,

you do not need to compile anything, you can just download the master and proceed as usual.
In the main Github page at the top: https://github.com/TT-ReBORN/Georgia-ReBORN 
click on the `<> Code button` > `Download zip`. [Or use this direct download link here.](https://github.com/TT-ReBORN/Georgia-ReBORN/archive/refs/heads/master.zip)

-TT

**Reply by azagthoth1** [reply]
Hello! I finally got to try it out, but there's the thing: As I've shown you in the previous screenshots, I like to have my library set up in a album-view mode, in which I'll usually just double-click them and go about my day, as I like to listen to albums (if I want to listen to random songs, I just double-click the "All (X items)" and turn the shuffle on), but I just realized that upon clicking the Add Tracks button, that instead of adding the track being currently played to Favorites, it is adding the whole album. Is there a way to avoid this? Am I doing something wrong or that's how it's gonna be since I'm playing albums? Having to play an album, switch to Library view, so I can select and play again the track I'd like to add to favorites and then click on Add Tracks kinda breaks the fun of the whole *I'm listening to an album* "I love this track!" *Clicks on Add Tracks button* and that's it.

I know beggars can't be choosers, you've done so much by adding this feature I suggested, but if there was a way to add only the track being currently played, even if I have the whole album on "Library view", to the favorites, would be perfect!

**Reply by TT-ReBORN** [reply]
@azagthoth1,

check the latest commit https://github.com/TT-ReBORN/Georgia-ReBORN/commit/8568ee92176f08ff5141917af9d29236a27e9990:
- Implemented AddTracks button functionality to add only the currently playing track #169
  * Instead of adding selected tracks, you can now add the currently playing track to your AddTracks playlist
  * To change the action: 
    - Go to the top Menu > Options > Player controls > Lower bar > Add tracks button > Now playing
    - Or right-click for context menu in the lower bar > Controls > Add tracks button > Now playing
    
 You can download the master and replace your files to use the new feature:
 https://github.com/TT-ReBORN/Georgia-ReBORN/archive/refs/heads/master.zip
 
 -TT
